unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,343 @@
+            logger.debug(f"Headers from HEAD request: {headers}")
+        except Exception as e:
+            logger.error(f"Failed to validate connection: {e}", exc_info=True)
+            raise SourceConnectionError(f"Failed to validate connection: {e}")
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        """Iterates over the GitHub repository tree, yielding `FileData` objects for all
+        files (blobs).
+
+        This method retrieves the entire repository tree for the specified branch or
+        the default branch.
+        For each file (blob), it extracts relevant metadata and yields a `FileData` object.
+
+        Args:
+            **kwargs (Any): Additional optional arguments.
+
+        Yields:
+            FileData: An object containing metadata and identifiers for each file in the repository.
+        """
+        repo = self.connection_config.get_repo()
+        sha = self.connection_config.branch or repo.default_branch
+        logger.info(f"Starting to look for blob files on GitHub in branch: {sha!r}")
+
+        git_tree = repo.get_git_tree(sha, recursive=True)
+
+        for element in git_tree.tree:
+            rel_path = element.path.replace(self.connection_config.repo_path, "").lstrip("/")
+            if element.type == "blob":
+                record_locator = {
+                    "repo_path": self.connection_config.repo_path,
+                    "file_path": element.path,
+                }
+                if self.connection_config.branch is not None:
+                    record_locator["branch"] = self.connection_config.branch
+
+                date_modified = datetime.strptime(
+                    element._headers["last-modified"],
+                    "%a, %d %b %Y %H:%M:%S %Z",
+                ).isoformat()
+
+                date_created = datetime.strptime(
+                    element._headers["date"],
+                    "%a, %d %b %Y %H:%M:%S %Z",
+                ).isoformat()
+
+                yield FileData(
+                    identifier=element.sha,
+                    connector_type=CONNECTOR_TYPE,
+                    source_identifiers=SourceIdentifiers(
+                        fullpath=element.path,
+                        filename=element.path.split("/")[-1],
+                        rel_path=rel_path,
+                    ),
+                    metadata=FileDataSourceMetadata(
+                        url=element.url,
+                        version=element.etag,
+                        record_locator=record_locator,
+                        date_modified=date_modified,
+                        date_created=date_created,
+                        date_processed=str(time()),
+                    ),
+                    additional_metadata={
+                        "content-type": element._headers.get("content-type", ""),
+                        "mode": element._rawData.get("mode", ""),
```

Comment:
Let's only set these fields in the `additional_metadata` dict if there's data to set rather than default to a blank string.